src/v/model/namespace.h

```diff
@@ -72,11 +72,18 @@ inline const model::ntp wasm_binaries_internal_ntp(
   model::topic("wasm_binaries"),
   model::partition_id(0));
 
+inline const model::topic
+  transform_log_internal_topic("_redpanda.transform_logs");
+
+inline const model::topic_namespace transform_log_internal_nt(
+  model::kafka_namespace, model::transform_log_internal_topic);
+
 inline bool is_user_topic(topic_namespace_view tp_ns) {
```

Comment:
TODO: understand the ramifications of this...

Reply:
yeah seriously...I'll write something up today or tomorrow

Reply:
Sorry this was for myself, but yeah I like @nvartolomei's suggestion to instead have topic properties for the stuff we want to enforce, so things like: `redpanda.canproduce`, `redpanda.candelete`, etc.

Reply:
I think the effects of this particular thing are even a bit more subtle, but yeah, I like that suggestion too.

Reply:
are we talking about the proliferation of special topic names and collections of names with a growing list of special behavior and traits.

Reply:
yes, i believe we are. this topic, the schemas topic, and the audit log topic all require special treatment. is this a leading question?

Reply:
> are we talking about the proliferation of special topic names and collections of names with a growing list of special behavior and traits.

Yes, and it's not clear what happens when one adds something to this list. Anyways, not for this PR, but something we should cleanup at somepoint.